### PR TITLE
404 page & name in signup

### PR DESCRIPTION
**Pull request by JosefForkman**
## Description
This PR add error pages and add name input on sign-up.  

## Changes
- **Error pages**
    - `error.tsx`
    - `[...not-found]/page.tsx`
    - `(sign in)/not-found.tsx`
    - `(sign in)/error.tsx`
- Fix tests
- **Name**
    - Add name to sign-up form component
    - Remove slug on sign-up form component
    - Handle user slug on API route sign up and update user

### CODE REVIEW

File: src/app/error.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+'use client'
+
+import Typography from '@/components/ui/Typography'
+import { Button } from '@/components/ui/button'
+import absurd from '../../public/absurd 2.png'
+import Image from 'next/image'
+import { useRouter } from 'next/navigation'
+
+export default function Error({
```
Comment (sebastianandreasson) [inline src/app/error.tsx:9]: why is error passed when not used?

**Review by sebastianandreasson — CHANGES_REQUESTED** [review]
Overall looks good, but it feels like the same component is duplicated 3 times right now. I would break this out into a component and reuse it everywhere. Right now if I want to change the text I need to do it 3 times.

File: src/app/error.tsx (hunk 1 above)
Comment (JosefForkman) [reply src/app/error.tsx:9]: It probably remained when I tried to figure out where error.tsx was supposed to be.

**Comment by JosefForkman** [comment]
Now is error and 400 pages to move to a component

**Review by sebastianandreasson — APPROVED** [review]

**Pull request merged**